examples/todomvc/app.re

```diff
@@ -36,7 +38,10 @@ module Top = {
       None
     };
     let handleChange {state} event =>
-      Some {...state, newTodo: ReasonJs.Document.value event##target};
+      switch (event##target |> Element.asHtmlElement) {
```

Comment:
asHtmlElement? No. But inheritance and lots and lots of interfaces are. And I don't really know how to model it idiomatically, so I've got lots of up- and downcasting methods. Including functors was one idea I mentioned in the reason-js PR.

Also, it gets worse. This is from reason-tools, with an Option module:
```reason
let appendToBody node =>
  Option.(
    document
      |> Document.asHtmlDocument
      |> and_then HtmlDocument.body
      |> map Element.asNode
      |> map (fun body => body |> Node.appendChild node)
  );
```

Equivalent to:
```javascript
document.body.appendChild(node);
```